pkg/reconciler/v1alpha1/taskrun/entrypoint/entrypoint.go

```diff
@@ -195,14 +195,26 @@ func getRemoteImage(image string, kubeclient kubernetes.Interface, build *buildv
 		return nil, fmt.Errorf("Failed to create k8schain: %v", err)
 	}
 
-	// this will first try to authenticate using the k8schain,
-	// then fall back to the google keychain,
-	// then fall back to anonymous
-	mkc := authn.NewMultiKeychain(kc, google.Keychain)
+	// this will first try to anonymous
+	// the fall back to authenticate using the k8schain,
+	// then fall back to the google keychain
+	mkc := authn.NewMultiKeychain(&anonymousKeychain{}, kc, google.Keychain)
 	img, err := remote.Image(ref, remote.WithAuthFromKeychain(mkc))
 	if err != nil {
 		return nil, fmt.Errorf("Failed to get container image info from registry %s: %v", image, err)
 	}
 
 	return img, nil
 }
+
+type anonymousKeychain struct{}
+
+func (a *anonymousKeychain) Resolve(_ name.Registry) (authn.Authenticator, error) {
+	return &anonymous{}, nil
```

Comment:
Does this return your own anonymous implementation since `NewMultiKeychain` has special logic to detect `authn.Anonymous`? ([ref](https://github.com/google/go-containerregistry/blob/5b94689e517344fe3474a719f6db9d9431ab40ab/pkg/authn/multikeychain.go#L40))

If so, can you document that, so I don't try to make this return `authn.Anonymous` in a future change?

Reply:
It returns our own anonymous implement **because** `NewMultiKeychain` has a special logic to detect `authn.Anonymous` (and will skip it until the end). Good point, I'll had a comment about that too :+1: